core/api/make.cpp

```diff
@@ -33,55 +85,202 @@ namespace fc::api {
         .ChainHead = {[&]() { return chain_store->heaviestTipset(); }},
         // TODO(turuslan): FIL-165 implement method
         .ChainNotify = {},
-        // TODO(turuslan): FIL-165 implement method
-        .ChainReadObj = {},
+        .ChainReadObj = {[&](const auto &cid) { return ipld->get(cid); }},
         .ChainTipSetWight = {[&](auto tipset_key)
                                  -> outcome::result<TipsetWeight> {
           OUTCOME_TRY(tipset, chain_store->loadTipset(tipset_key));
           return weight_calculator->calculateWeight(tipset);
         }},
         // TODO(turuslan): FIL-165 implement method
         .MarketEnsureAvailable = {},
-        // TODO(turuslan): FIL-165 implement method
-        .MinerCreateBlock = {},
+        .MinerCreateBlock = {[&](auto &miner,
+                                 auto &parent,
+                                 auto &ticket,
+                                 auto &proof,
+                                 auto &messages,
+                                 auto height,
+                                 auto timestamp) -> outcome::result<BlockMsg> {
+          OUTCOME_TRY(context, tipsetContext(parent, true));
+
+          OUTCOME_TRY(miner_state, context.minerState(miner));
+          OUTCOME_TRY(worker_id,
+                      context.state_tree.lookupId(miner_state.info.worker));
+          OUTCOME_TRY(state_root, context.state_tree.flush());
+
+          BlockMsg block;
+          block.header.miner = miner;
+          block.header.parents = parent.cids;
+          block.header.ticket = ticket;
+          block.header.height = height;
+          block.header.timestamp = timestamp;
+          block.header.epost_proof = proof;
+          block.header.parent_state_root = state_root;
+          block.header.parent_message_receipts = *context.receipts;
```

Comment:
at least add assert that receipts field has a value

Reply:
`tipsetContext(parent, true)` sets `receipts`, so it has value